### ISSUE TITLE
Renaming a Mode throws ClassCastException

### ISSUE THREAD

**Issue by philip-alldredge** [osate/osate2#1449]
## Summary
When renaming a mode with the graphical editor, a ClassCastException is thrown if that mode is referenced by a flow specification. We believe it is due to a problem with org.osate.aadl2.operations.ModalPathOperations#getInModes not returning the type expected by EMF. This is very similar to #1021.

## Steps to Reproduce
1. Create diagram for model below.
2. Rename m1 to m1_renamed using the graphical editor.
3. A window will appear reporting the error and indicating that the rename participant provided by the graphical editor will not be used in the future.
[Exception.txt](https://github.com/osate/osate2/files/2288057/Exception.txt)

```aadl
package bug
public
	system top
		features
			i1: in data port;
		flows
			fsnk: flow sink i1 in modes (m1);
		modes
			m1: mode;
	end top;
end bug;
```

## Environment

* **OSATE Version**: develop
* **Operating System**:  Windows 10

**Comment by lwrage** [comment]
Need to check if there are more usages of `BasicInternalEList` that need to be replaced.

**Comment by philip-alldredge** [comment]
#1489 
1455

**Comment by philip-alldredge** [comment]
The fix for this(#1455) caused additional issues. Similar to the way the fix for #1021 caused a write transaction error, this can cause a write transaction error in certain circumstances. @RyanMcilnay will provide additional details.

**Comment by RyanMcilnay** [comment]
While editing a model, I received a write transaction error while the model was being serialized. It was triggered by creating a classifier implementation with Modes and ModalElements, then selecting a mode from the the highlight-in-mode drop-down in the toolbar.

I have attached the stack trace.
[IllegalStateException_Stack_Trace.txt](https://github.com/osate/osate2/files/2337704/IllegalStateException_Stack_Trace.txt)
